Rss ile Tüm Haberleri Cekip Database' e kaydetme

yunuscanbartik opened the pull request.

yunuscanbartik committed:
Add NewsCatcherBackgroundService for BBC RSS fetching

Introduced a new background service project, NewsCatcherBackgroundService, to periodically fetch and process BBC Turkish RSS feed items. Implemented IRssFeedService and its RssFeedService implementation, added BBC RSS feed model classes for XML deserialization, and registered the background worker in Program.cs. Included NLog for logging and provided configuration files for RSS feed URL and environment-specific settings. Updated the solution file to include the new project and its build configurations.

yunuscanbartik committed:
Rss den aldığım verileri ona özgü olan modele kaydettikten sonra bir fonksiyon yazdım ve bu fonksiyonda kedi modelimle eşledim.

yunuscanbartik committed:
Add database saving for RSS items and refactor background svc

yunuscanbartik committed:
Refactor BBCModel and update RSS feed service interfaces

yunuscanbartik committed:
Revert "Refactor BBCModel and update RSS feed service interfaces"

This reverts commit 65632d2a2ca8df31ef7a984e0f5954a5ce87e153.

yunuscanbartik committed:
Refactor RSS model, update service interfaces, improve logging

Refactored the BBC RSS feed model in NewsModel.cs for simpler XML mapping and custom date parsing. Updated IRssFeedService interface methods to use CreateModel.ReturnData. Changed database connection string and added RabbitMQ settings in appsettings.json. Registered IRssFeedService as singleton and enabled console logging in Program.cs. Refactored RssBackgroundService for periodic RSS fetching with better error handling and logging. Updated RssFeedService methods to use the new data model and improved logging.

yunuscanbartik committed:
Clear SQL parameters at start of each foreach iteration

yunuscanbartik committed:
Improve DB null handling and fix NewsService read logic

atakancelik commented on NewsCatcher.Models/Models/NewsModel.cs:
```diff
@@ -104,5 +106,90 @@ public class ReturnData
+                public string Link { get; set; }
+
+                [XmlElement("lastBuildDate")]
+                public string LastBuildDateStr { get; set; }
```
Hepsi nullable olsa iyi olur xml parser sorun yaratabilir.

yunuscanbartik replied:
hepsi nullable yapıldı

atakancelik commented on NewsCatcher.Models/Models/NewsModel.cs:
```diff
@@ -104,5 +106,90 @@ public class ReturnData
+                [XmlElement("guid")]
+                public string Guid { get; set; }
+
+                [XmlElement("thumbnail", Namespace = "http://search.yahoo.com/mrss/")]
```
Nedir

yunuscanbartik replied:
bu GuId ye göre redisde key value işlemi yapılacak.

yunuscanbartik replied:
Thumbnail yanındaki namespace ise silindi. normalde bu da haberlerin fotoğrafa basıldığında yönlendiği link olarak tutulacaktır.

atakancelik commented on NewsCatcher.Services/Services/NewsService.cs:
```diff
@@ -31,19 +31,22 @@ public NewsService(IDatabaseContext dbContext)
+                                Title = reader.GetString("Title"),
+                                Content = reader.GetString("Content"),
+                                Summary = reader.GetString("Summary"),
+                                CategoryId = reader["CategoryId"] == DBNull.Value ? (int?)null : reader.GetInt32(reader.GetOrdinal("CategoryId")),
```
Bu kontrol her alanda olabilir, fonksiyona çevirebilirsin.

atakancelik commented on NewsCatcherBackgroundService/RssFeedService.cs:
```diff
@@ -0,0 +1,124 @@
+
+                    var xmlContent = await response.Content.ReadAsStringAsync(); //response içeriğini string olarak okuyorum. 
+
+                    var serializer = new XmlSerializer(typeof(NewsModel.BBCModel.Rss)); // xml i parçalayarak nesneye dönüştürmek için XmlSerializer kullanıyorum.
```
Feed urle göre nasıl bbc dedik switchcase vs mevcut değil

yunuscanbartik replied:
sadece bbc mevcut şuan

atakancelik commented on NewsCatcherBackgroundService/RssFeedService.cs:
```diff
@@ -0,0 +1,124 @@
+            {
+                foreach (var item in returnDataList)
+                {
+                    sqlCommand.Parameters.Clear();
```
clear yerine newlemek daha sağlıklı.

yunuscanbartik committed:
clear kaldırıldı ve using ile newlenerek kullanıldı.

yunuscanbartik committed:
Add ThumbnailUrl, Link, GuId to News models and services

Extend NewsModel classes and related services to support ThumbnailUrl, Link, and GuId properties for news items. Update NewsService and RssFeedService to handle these fields when reading, inserting, and updating news records. Improve CategoryId handling in RssFeedService to use DBNull.Value when unset. These changes enable richer metadata storage and processing for news articles.

atakancelik reviewed (approved):
LGTM
